sources/packages/api/src/route-controllers/student/models/student.dto.ts

```diff
@@ -181,3 +189,31 @@ export class SINValidationsAPIOutDTO {
   temporarySIN: boolean;
   sinExpiryDate?: string;
 }
+
+/**
+ * DTO to allow manually creation of SIN validations.
+ */
+export class CreateSINValidationAPIInDTO {
+  @IsValidSIN()
```

Comment:
Nice validator

Reply:
@dheepak-aot @henryoforeh-aot just to share a conversation that I had with @guru-aot, this code is also used directly on form.io definitions that contain a SIN to be validated. It was kept simple and mostly the same as the one used in form.io. Please keep in mind also that the code does not use any of the newest javascript features to keep it as compatible as possible.